src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1800,7 +1812,12 @@ public ImmutableArray<DocumentId> GetDocumentIdsWithFilePath(string? filePath)
         private ImmutableDictionary<ProjectId, CompilationTracker> CreateCompilationTrackerMap(ProjectId projectId, ProjectDependencyGraph dependencyGraph)
         {
             var builder = ImmutableDictionary.CreateBuilder<ProjectId, CompilationTracker>();
-            var dependencies = dependencyGraph.GetProjectsThatTransitivelyDependOnThisProject(projectId);
+            var allowReuse = true;
+            if (!dependencyGraph.TryGetProjectsThatTransitivelyDependOnThisProject(projectId, out var dependencies))
```

Comment:
Interesting. The issue here is we don't want to force an expensive computation (transitive reverse dependencies) to avoid  a cheap computation (`tracker.Fork`). The one thing I haven't tried yet is only calculating `GetProjectsThatTransitivelyDependOnThisProject` _after_ `tracker.HasCompilation` returns true.

Reply:
Updated to no longer drop compilations if they can be reused